great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -325,8 +326,14 @@ def _init_factories(self) -> None:
 
         self._checkpoints: CheckpointFactory | None = None
         if checkpoint_store := self.stores.get(self.checkpoint_store_name):
+            # NOTE: Currently in an intermediate state where both the legacy and V1 stores exist.
+            #       Upon the deletion of the old checkpoint store, the new one will be promoted.
+            v1_checkpoint_store = V1CheckpointStore()
+            v1_checkpoint_store._store_backend = (
+                checkpoint_store.store_backend
+            )  # Leverage same backend as what was configured for the old store
```

Comment:
this is probably the best/only way to do this right now, but just noting that this is an example of how our design is broken.

Reply:
Yeah this feels gross. The lift to actually completely remove old checkpoints was far too cumbersome so we're going to be in a weird intermediate state where both exist (with the legacy ones only being around for test coverage).